go/ql/test/library-tests/semmle/go/Types/Aliases.ql

```diff
@@ -1,4 +1,10 @@
 import go
 
+string getQualifiedNameIfExists(Type t) {
+  if exists(t.getQualifiedName()) then result = t.getQualifiedName() else result = t.getName()
```

Comment:
I'm not aware of any examples of types that don't have `getQualifiedName`. Is this related to alias types? We should consider updating `getQualifiedName` if it makes sense. It has a special case already to deal with built-in functions like `make`, which don't have a package.

Reply:
While writing this test, I wasn't getting results without this check. I think specifically `string` and other base types didn't have a qualified name.

Reply:
I've looked into this now and I agree that the test does need something like this. We basically want fully qualified names for named types and for other types we probably want `pp()` (many types don't have `getName()`, like `func(T) T` or `[10]T`).